### PR TITLE
Initial Windows port for K3s multi-exec binary

### PR DESCRIPTION
**Pull request by seanyen**
* Initial Windows port for the multi-exec binary bring up.

### CODE REVIEW

**Review by manuelbuil — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -1,35 +1,5 @@
...
-	reexec.Register("firewall", firewall.Main)
-	reexec.Register("flannel", flannel.Main)
-	reexec.Register("host-local", hostlocal.Main)
-	reexec.Register("loopback", loopback.Main)
-	reexec.Register("portmap", portmap.Main)
-	if !reexec.Init() {
-		_, _ = fmt.Fprintln(os.Stderr, bv.BuildString("plugins"))
-		_, _ = fmt.Fprintf(os.Stderr, "CNI protocol versions supported: %s\n", strings.Join(version.All.SupportedVersions(), ", "))
-	}
+	mainEntry()
```
Comment (brandond) [inline main.go]: I think you could do away this this indirection entirely, and just have a distinct`func main() {}` for linux and windows.

**Review by brandond — CHANGES_REQUESTED** [review]
one nit on the linux/windows main split, lgtm otherwise!

File: main.go (hunk 1 above)
Comment (seanyen) [reply main.go]: @brandond I removed the `main.go` in this case, and just keep `func main() {}` in both platform specific files. Let me know if I capture the idea correctly. thanks!

File: main_windows.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/pkg/reexec"
+
+	"github.com/containernetworking/cni/pkg/version"
+	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+	hostlocal "github.com/containernetworking/plugins/plugins/ipam/host-local"
+	"github.com/containernetworking/plugins/plugins/main/windows/win-bridge"
+    "github.com/containernetworking/plugins/plugins/main/windows/win-overlay"
```
Comment (brandond) [inline main_windows.go]: nit: whitespace. Are you using an editor that supports autoformatting? If not, you can use `go fmt` to clean it up for you.
```suggestion
	"github.com/containernetworking/plugins/plugins/main/windows/win-overlay"
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: main_windows.go (hunk 2 above)
Comment (seanyen) [reply main_windows.go]: @brandond thanks for catching that! I think my editor is using spaces by default. The formatting should be corrected now.

**Review by brandond — APPROVED** [review]

**Pull request merged**